### DISCUSSION TITLE
ESP32C3 Support?

### DISCUSSION THREAD

**Discussion by JohnMacrae** [UncleRus/esp-idf-lib#173] [Ideas]
Hi and thanks for providing this resource :)

I was wondering if there are any updates planned to allow for use for the ESP32C3?

**Comment by UncleRus** [comment]
Hi!
I'm not sure about official ESP32C3 support in esp-idf-lib (I don't have this chip, so I can't test and debug), but @BugerDread has implemented this (somehow hacky) in https://github.com/BugerDread/esp-idf-lib/tree/esp32-c3-hack

**Comment by JohnMacrae** [comment]
Many thanks for the link! Once I get over a few bumps in the road, hopefully I may be able to contribute.

**Comment by trombik** [comment]
if one wants to add riscv32 support to `esp-idf-lib`, `components/esp_idf_lib_helpers` should be patched.

p.s. for the record, the source code of the toolchain, `riscv32-esp-elf`, is not open to public while the binary is (i'm not a linux user).

**Comment by UncleRus** [comment]
@JohnMacrae I added support for esp32c3 in the master. Could you test it?

**Comment by JohnMacrae** [comment]
Yes - I will - thank you very much.  I can't do it for a few days but I
will get to it :)

Kind Regards
John

**Comment by UncleRus** [comment]
@JohnMacrae Any news about testing?

**Comment by JohnMacrae** [comment]
@UncleRus/esp-idf-lib
***@***.***>
My apologies!  Yes - I tested with the DS18b20 and it all went well - thank
you so much :)

Kind Regards
John
